monkey/infection_monkey/island_api_client/island_api_client_errors.py

```diff
@@ -0,0 +1,30 @@
+class IslandAPIError(Exception):
+    """
+    Raised when something goes wrong when calling the Island API
+    """
+
+    pass
+
+
+class IslandAPITimeoutError(IslandAPIError):
+    """
+    Raised when the API request hits a timeout
+    """
+
+    pass
+
+
+class IslandAPIConnectionError(IslandAPIError):
+    """
+    Raised when the API request can't find/connect to the Island
+    """
+
+    pass
+
+
+class IslandAPIRequestFailedError(IslandAPIError):
```

Comment:
This is combining two different things that probably shouldn't be combined:
1. The client made a mistake (malformed request, unknown endpoint, stuff like that)
2. Something went wrong on the server

Reply:
I think `IslandAPIError` makes sense for 2.

Reply:
We probably just need to update the docstring.